src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,842 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+        if (statChangeAtk < -6) {
+            statChangeAtk = -6;
+        }
+        if (statChangeAtk > 6) {
+            statChangeAtk = 6;
+        }
+        statChangeSpAtk *= 2;
+        if (statChangeSpAtk < -6) {
+            statChangeSpAtk = -6;
+        }
+        if (statChangeSpAtk > 6) {
+            statChangeSpAtk = 6;
+        }
+    }
+    
+    if (CheckBattlerAbilityIfNotIgnored(ctx, battlerIdAttacker, battlerIdTarget, ABILITY_SIMPLE) == TRUE) {
+        statChangeDef *= 2;
+        if (statChangeDef < -6) {
+            statChangeDef = -6;
+        }
+        if (statChangeDef > 6) {
+            statChangeDef = 6;
+        }
+        statChangeSpDef *= 2;
+        if (statChangeSpDef < -6) {
+            statChangeSpDef = -6;
+        }
+        if (statChangeSpDef > 6) {
+            statChangeSpDef = 6;
+        }
+    }
+    
+    if (CheckBattlerAbilityIfNotIgnored(ctx, battlerIdAttacker, battlerIdTarget, ABILITY_UNAWARE) == TRUE) {
+        statChangeAtk = 0;
+        statChangeSpAtk = 0;
+    }
+    if (calcAttacker.ability == ABILITY_UNAWARE) {
+        statChangeDef = 0;
+        statChangeSpDef = 0;
+    }
+    
+    statChangeAtk += 6;
+    statChangeDef += 6;
+    statChangeSpAtk += 6;
+    statChangeSpDef += 6;
+    
+    if (calcAttacker.ability == ABILITY_RIVALRY && calcAttacker.gender == calcTarget.gender && calcAttacker.gender != 2 && calcTarget.gender != 2) {
+        movePower = movePower * 125 / 100;
+    }
+    if (calcAttacker.ability == ABILITY_RIVALRY && calcAttacker.gender != calcTarget.gender && calcAttacker.gender != 2 && calcTarget.gender != 2) {
+        movePower = movePower * 75 / 100;
+    }
+    
+    for (i = 0; i < NELEMS(sPunchingMoves); i++) {
+        if (sPunchingMoves[i] == moveNo && calcAttacker.ability == ABILITY_IRON_FIST) {
+            movePower = movePower * 12 / 10;
+            break;
+        }
+    }
+    
+    if (!CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_CLOUD_NINE) && !CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_AIR_LOCK)) {
+        if ((fieldCondition & FIELD_CONDITION_SUN_ALL) && calcAttacker.ability == ABILITY_SOLAR_POWER) {
+            monSpAtk = monSpAtk * 15 / 10;
+        }
+        if ((fieldCondition & FIELD_CONDITION_SANDSTORM_ALL) && (calcTarget.type1 == TYPE_ROCK || calcTarget.type2 == TYPE_ROCK)) {
+            monSpDef = monSpDef * 15 / 10;
+        }
+        if ((fieldCondition & FIELD_CONDITION_SUN_ALL) && CheckAbilityActive(bsys, ctx, CHECK_ABILITY_SAME_SIDE_HP, battlerIdAttacker, ABILITY_FLOWER_GIFT)) {
+            monAtk = monAtk * 15 / 10;
+        }
+        if ((fieldCondition & FIELD_CONDITION_SUN_ALL) && GetBattlerAbility(ctx, battlerIdAttacker) != ABILITY_MOLD_BREAKER && CheckAbilityActive(bsys, ctx, CHECK_ABILITY_SAME_SIDE_HP, battlerIdTarget, ABILITY_FLOWER_GIFT)) {
+            monSpDef = monSpDef * 15 / 10;
+        }
+    }
+    
+    if (ctx->unk_334.moveData[moveNo].effect == 7) {
```

Comment:
do we know what the values for effect are?

Reply:
I think I know the one for explosion but none of the other ones